### PR TITLE
feat(twap-orders): No. of parts field (2)

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

This PR adds a `No. of parts` input field
The code structure and functionality will be updated in the future PR's
Doing this as a separate PR for a simpler review.
The second repeated Field will be slippage in the following PR.

We can skip QA review for now on this.

# To test
1. Open dev tools
2. In console paste `localStorage.setItem('advanced-orders', '1')` and press enter then reload page
3. You should now see Advanced tab

### CODE REVIEW

File: src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  updateAdvancedOrdersSettingsAtom,
+} from '@cow/modules/advancedOrders/state/advancedOrdersSettingsAtom'
+
+export function NumberOfParts() {
+  const { numberOfParts } = useAtomValue(advancedOrdersSettingsAtom)
+
+  const updateSettingsState = useSetAtom(updateAdvancedOrdersSettingsAtom)
+
+  const onUserInput = (value: string) => {
+    updateSettingsState({ numberOfParts: Number(value) })
```
Comment (alfetopito) [inline src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx:17]: What happens if `value` is not a number?

File: src/cow-react/modules/advancedOrders/containers/AdvancedOrdersWidget/index.tsx [hunk 2]
```diff
@@ -52,6 +54,11 @@ export function AdvancedOrdersWidget() {
     settingsWidget: <div></div>,
     bottomContent: (
       <>
+        <styledEl.Row>
+          <NumberOfParts />
+          <NumberOfParts />
```
Comment (alfetopito) [inline src/cow-react/modules/advancedOrders/containers/AdvancedOrdersWidget/index.tsx:59]: Is it intentional to add it twice?

**Review by alfetopito — COMMENTED** [review]
How can I see the form?
Can you please add instructions to access it?

**Comment by nenadV91** [comment]
@alfetopito "How can I see the form?"
Sorry about that, I will add the instructions in description.

File: src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx (hunk 1 above)
Comment (alfetopito) [reply src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx:17]: I was not able to insert anything that's not an integer, so I guess this is covered in the component level

**Review by alfetopito — APPROVED** [review]
Thank you for the instructions Nenad!

Looking good.

The only comment is the placeholder 
image

Since it only accepts integers, it should not show a decimal by default to not give the wrong impression

File: src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx:17]: it doesn't harm to do as you did in another PR and add a default or something, so you don't get a NaN

**Review by anxolin — APPROVED** [review]
Not for this PR, but for the refactor and new types of orders i think might be good to use a bit more cosmos

File: src/cow-react/modules/advancedOrders/containers/AdvancedOrdersWidget/index.tsx (hunk 2 above)
Comment (alfetopito) [reply src/cow-react/modules/advancedOrders/containers/AdvancedOrdersWidget/index.tsx:59]: It's used as a placeholder. In the next PR it is replaced by the slippage component

File: src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/cow-react/modules/advancedOrders/containers/NumberOfParts/index.tsx:17]: + What about decimals? I can't input `12.95` (we need it for slippage value)

**Pull request merged**